From 45d2d5173aa1cd27ef15180b4afe3e4779d26019 Mon Sep 17 00:00:00 2001
From: Jim Stichnoth <jstichnoth@mythtv.org>
Date: Tue, 5 May 2015 21:12:59 -0700
Subject: [PATCH] Subtitles: Fix cc608 preamble indents.  Fixes #12051.

Cherry-picked from 366c428708682420bbf53a301d145ca441aedbdd .

Thanks to Helen (faginbagin) for the analysis and patch.

As Helen points out in the ticket, it would be all-around better if we
took an approach like the cc708 captions and rendered everything onto
an internal grid of characters and then converted that to strings to
draw on the screen, but that's a lot of work for a mostly deprecated
caption format.
---
 mythtv/libs/libmythtv/cc608decoder.cpp   |    4 ++--
 mythtv/libs/libmythtv/subtitlescreen.cpp |    7 -------
 2 files changed, 2 insertions(+), 9 deletions(-)

diff --git a/mythtv/libs/libmythtv/cc608decoder.cpp b/mythtv/libs/libmythtv/cc608decoder.cpp
index f9768c1..035ea42 100644
--- a/mythtv/libs/libmythtv/cc608decoder.cpp
+++ b/mythtv/libs/libmythtv/cc608decoder.cpp
@@ -323,6 +323,7 @@ void CC608Decoder::FormatCCField(int tc, int field, int data)
                                     .arg(b2, 2, 16));
                             // Encode as 0x7000 through 0x700f for the
                             // 16 possible values of b2.
+                            ccbuf[mode] += ' ';
                             ccbuf[mode] += QChar(0x7000 + (b2 & 0xf));
                             len = ccbuf[mode].length();
                             col[mode]++;
@@ -690,7 +691,7 @@ QString CC608Decoder::ToASCII(const QString &cc608str, bool suppress_unknown)
                 if (cpu >= 0x7000 && cpu < 0x7000 + 0x30)
                 {
                     if (!suppress_unknown)
-                        ret += QString("[%1]").arg(cpu - 0x7000, 2, 16);
+                        ret += QString("[%1]").arg(cpu, 2, 16);
                 }
                 else if (cpu <= 0x80)
                     ret += QString(cp.toLatin1());
@@ -856,7 +857,6 @@ int CC608Decoder::NewRowCC(int mode, int len)
     {
         ccbuf[mode] += QChar(newattr[mode] + 0x7000);
         len++;
-        col[mode]++;
     }
 
     newcol[mode] = 0;
diff --git a/mythtv/libs/libmythtv/subtitlescreen.cpp b/mythtv/libs/libmythtv/subtitlescreen.cpp
index a3863f6..b48fb9c 100644
--- a/mythtv/libs/libmythtv/subtitlescreen.cpp
+++ b/mythtv/libs/libmythtv/subtitlescreen.cpp
@@ -1311,13 +1311,6 @@ static QString extract_cc608(QString &text, int &color,
     else
     {
         result = text.left(nextControl);
-        // Print the space character before handling the next control
-        // character, otherwise the space character will be lost due
-        // to the text.trimmed() operation in the MythUISimpleText
-        // constructor, combined with the left-justification of
-        // captions.
-        if (text[nextControl] < (0x7000 + 0x10))
-            result += " ";
         text = text.mid(nextControl);
     }
 
-- 
1.7.10.2

